### PR TITLE
Feature/menu bar component

### PR DESCRIPTION
**Pull request by vikasrohit**
PR for #10 
@nlitwin @samsep @aselbie fyi, please let me know if I am doing anything out of standards.

### CODE REVIEW

File: components/MenuBar/MenuBar.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+class MenuBar extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {mobile: false}
+  }
+
+  componentWillMount() {
+    this.handleResize(this)
+    const handleResize = this.handleResize
+    const thisRef = this
```
Comment (nlitwin) [inline components/MenuBar/MenuBar.jsx]: We can use `.bind()` or ES6 arrow functions to preserve the correct `this` value.

File: components/MenuBar/MenuBar.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { PropTypes, Component } from 'react'
+
+require('./MenuBar.scss')
+
+class MenuBar extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {mobile: false}
+  }
```
Comment (nlitwin) [inline components/MenuBar/MenuBar.jsx]: If we add `this.handleResize = this.handleResize.bind(this)` in the constructor, we can remove the code in `componentWillMount()` and just leave `window.addEventListener('resize', this.handleResize)`. I believe we don't need to put the `handleResize` method on `this.state`

File: components/MenuBar/MenuBar.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    this.state.handler = function() { handleResize(thisRef) }
+    window.addEventListener('resize', this.state.handler)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.state.handler)
+  }
+
+  handleResize(thisRef) {
+    let breakPoint = thisRef.props.mobileBreakPoint
```
Comment (nlitwin) [inline components/MenuBar/MenuBar.jsx]: We should also be able to remove `thisRef` as an argument to `handleResize` and just use `this.props.propertyName`

File: components/MenuBar/MenuBar.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  render() {
+    let mbClasses = 'MenuBar'
+    let orientation = this.props.orientation
+    if (!orientation || ['horizontal', 'vertical'].indexOf(orientation) === -1) {
+      orientation = 'horizontal'
+    }
+    mbClasses += (' ' + orientation)
+    return (
+      <ul className={mbClasses}>
```
Comment (nlitwin) [inline components/MenuBar/MenuBar.jsx]: Not sure if it's necessary here, but I've found the `classnames` npm package really useful. https://github.com/JedWatson/classnames

File: components/MenuBar/MenuBar.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    )
+  }
+}
+
+MenuBar.propTypes = {
+  items             : PropTypes.array.isRequired,
+  mobileBreakPoint  : PropTypes.number
+}
+
+MenuBar.MobileBreakPoint = 768
```
Comment (nlitwin) [inline components/MenuBar/MenuBar.jsx]: Is this a default prop? If so I think we need something like:
`MenuBar.defaultProps = { mobileBreakPoint: 768 }`

File: components/MenuBar/MenuBar.jsx (hunk 1 above)
Comment (vikasrohit) [inline components/MenuBar/MenuBar.jsx]: Thanks @nlitwin I was wondering how can do that and when I didn't any way that time, I used thisRef. :) I would fix it.

File: components/MenuBar/MenuBar.jsx (hunk 5 above)
Comment (vikasrohit) [inline components/MenuBar/MenuBar.jsx]: Yes, it is. Thanks for pointing it.

**Comment by vikasrohit** [comment]
@nlitwin I have fixed most of  the suggestions except the classnames. May be we can fix it across components in one go.

**Pull request merged**